src/api/app/components/workflow_run_filter_component.html.haml

```diff
@@ -21,42 +21,69 @@
+                                                            checked: @selected_filter[:status]&.include?('fail')}
   .list-group.list-group-flush.mt-4.mb-4
     %h6.px-3.py-2
       %b Event Type
       = render partial: 'webui/shared/info_popover', locals: { position: 'bottom', text: event_type_info_text }
-    = render WorkflowRunFilterCheckboxComponent.new(text: 'Pull/Merge Request', amount: @count['pull_request'],
-                                                filter_item: { generic_event_type: 'pull_request' },
-                                                selected_filter: @selected_filter.slice(:event_type),
-                                                key: 'pull_request')
-    = render WorkflowRunFilterCheckboxComponent.new(text: 'Push', amount: @count['push'],
-                                                filter_item: { generic_event_type: 'push' },
-                                                selected_filter: @selected_filter.slice(:event_type))
-    = render WorkflowRunFilterCheckboxComponent.new(text: 'Tag Push', amount: @count['tag_push'],
-                                                filter_item: { generic_event_type: 'tag_push' },
-                                                selected_filter: @selected_filter.slice(:event_type))
+    .px-4.py-1
+      = render partial: 'webui/shared/check_box', locals: { label: 'Pull/Merge Request',
+                                                            amount: @count['pull_request'] || 0,
+                                                            key: 'pull_request',
+                                                            checked: @selected_filter[:event_type]&.include?('pull_request')}
+    .px-4.py-1
+      = render partial: 'webui/shared/check_box', locals: { label: 'Push',
+                                                            amount: @count['push'] || 0,
+                                                            key: 'push',
+                                                            checked: @selected_filter[:event_type]&.include?('push')}
+    .px-4.py-1
+      = render partial: 'webui/shared/check_box', locals: { label: 'Tag Push',
+                                                            amount: @count['tag_push'] || 0,
+                                                            key: 'tag_push',
+                                                            checked: @selected_filter[:event_type]&.include?('tag_push')}
+
   .list-group.list-group-flush.mt-4.mb-4
     %h6.px-3.py-2
       %b Event Action
       = render partial: 'webui/shared/info_popover', locals: { position: 'bottom', text: pr_mr_request_info_text }
-    = render WorkflowRunRequestActionFilterComponent.new(filter_item: 'request_action', selected_filter: @selected_filter[:request_action])
+    .px-4.py-1
+      - request_action_options = ['all'] + SCMWebhook::ALLOWED_PULL_REQUEST_ACTIONS + SCMWebhook::ALLOWED_MERGE_REQUEST_ACTIONS
```

Comment:
Maybe this can go inside SCMWebhook as well

Reply:
Makes sense. Fixed